File: ERP-INVESTIGACION/articulos-frontend/src/app/features/configuraciones/services/clinica.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Clinica, ClinicaAuditoria } from '../models/empresa.model';
import { ApiResponse } from '../../../shared/modals/response.model';

@Injectable({
    providedIn: 'root'
})
export class ClinicaService {
    private apiUrl = `${environment.apiUrl}/clinica`;

    constructor(private http: HttpClient) { }

    obtenerPrincipal(): Observable<ApiResponse<Clinica>> {
        return this.http.get<ApiResponse<Clinica>>(this.apiUrl);
    }

    obtenerPorId(id: number): Observable<ApiResponse<Clinica>> {
        return this.http.get<ApiResponse<Clinica>>(`${this.apiUrl}/${id}`);
    }

    obtenerHistorial(idClinica: number): Observable<ApiResponse<any>> {
        return this.http.get<ApiResponse<any>>(`${this.apiUrl}/historial/${idClinica}?page=0&size=1000`);
    }

    crear(clinica: Clinica): Observable<ApiResponse<Clinica>> {
        return this.http.post<ApiResponse<Clinica>>(this.apiUrl, clinica);
    }

    actualizar(id: number, clinica: Clinica): Observable<ApiResponse<Clinica>> {
        return this.http.put<ApiResponse<Clinica>>(`${this.apiUrl}/${id}`, clinica);
    }

    eliminar(id: number): Observable<ApiResponse<void>> {
        return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/${id}`);
    }
}

File: ERP-INVESTIGACION/articulos-backend/src/main/java/com/pe/articulos/modules/clinica/service/ClinicaService.java
package com.pe.articulos.modules.clinica.service;

import com.pe.articulos.modules.clinica.entity.Clinica;
import com.pe.articulos.modules.clinica.entity.ClinicaAuditoria;
import org.springframework.data.domain.Pageable;
import com.pe.articulos.core.shared.dto.PageResponse;

public interface ClinicaService {
    Clinica obtenerPrincipal();

    Clinica obtenerPorId(Long id);

    Clinica crear(Clinica clinica);

    Clinica actualizar(Long id, Clinica clinica);

    void eliminar(Long id);

    PageResponse<ClinicaAuditoria> listarHistorial(Long idClinica, Pageable pageable);
}
